store/src/java/com/zimbra/cs/imap/ImapCredentials.java

```diff
@@ -159,7 +160,8 @@ private void saveSubscriptions(Set<String> subscriptions) throws ServiceExceptio
     }
 
     protected Set<String> listSubscriptions() throws ServiceException {
-        return getImapMailboxStore().listSubscriptions(getContext());
+        Set<String> subs = getImapMailboxStore().listSubscriptions(getContext());
```

Comment:
not sure why this this change is needed

Reply:
I'll add comments for clarity and squash the commit with the previous change.  This is actually the cause of the user visible problem addressed by the bug - was later trying to modify the returned set when subscribing/unsubscribing and that doesn't go too well when the set is not modifiable.